game_server/src/collide.rs

```diff
@@ -0,0 +1,74 @@
+use crate::maths::{Vec2, Box, RBox};
+
+pub trait Collide<Rhs> {
+    fn collides(&self, other: &Rhs) -> bool;
+}
+
+impl Collide<Vec2> for Vec2 {
+    fn collides(&self, other: &Self) -> bool {
+        self == other
+    }
+}
+
+impl Collide<Vec2> for Box {
+    fn collides(&self, other: &Vec2) -> bool {
+        self.pos < *other && other < &(self.pos + self.size) 
+    }
+}
+
+impl Collide<Box> for Box {
+    fn collides(&self, other: &Self) -> bool {
+        self.collides(&other.pos)
+        || self.collides(&(other.pos + Vec2{x: other.size.x, y: 0.0}))
+        || self.collides(&(other.pos + Vec2{x: 0.0, y: other.size.y}))
+        || self.collides(&(other.pos + other.size))
+
+        || other.collides(&(self.pos))
+        || other.collides(&(self.pos + Vec2{x: self.size.x, y: 0.0}))
+        || other.collides(&(self.pos + Vec2{x: 0.0, y: self.size.y}))
+        || other.collides(&(self.pos + self.size))
+    }
+}
+
+impl Collide<Vec2> for RBox {
+    fn collides(&self, other: &Vec2) -> bool {
```

Comment:
This function should probably be (completely) re-thinked (especially the manhattan-thing xd) (or the structure of RBoxes)

Reply:
RBoxes are very usefull for simple hitbok modelling